### PR TITLE
Add simple exclude all columns test to sqllogictest

### LINKED ISSUE #6510: SELECTing 0 column with EXCLUDE gives projection_push_down error
### Describe the bug

Assume we have a table named `table1`  with single column `a` such as below
|a|
|:----:|
|1|
|2|
|3|
|4|

when I run the query below on this table
```sql
SELECT * EXCEPT(a)
FROM table1
```
It gives `Error: Context("push_down_projection", Internal("Optimizer rule 'push_down_projection' failed, due to generate a different schema, original schema: DFSchema { fields: [], metadata: {} }, new schema: DFSchema { fields: [DFField { qualifier: Some(Bare { table: \"table1\" }), field: Field { name: \"a\", data_type: Int32, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} } }], metadata: {} }"))` error.

However, assume we have an empty table with no columns named `empty_table`. Query below works
```sql
SELECT *
FROM empty_table
```
So I don't think bug is related to showing empty columns, but I am not sure.

### To Reproduce

You can use test below to reproduce problem
```rust
#[tokio::test]
async fn test_exclude_error() -> Result<()> {
    let config = SessionConfig::new()
        .with_target_partitions(1);
    let ctx = SessionContext::with_config(config);
    ctx.sql("CREATE TABLE table1 (
          a INT,
        ) as VALUES
              (1),
              (2),
              (3),
              (4)").await?;

    let sql = "SELECT * EXCEPT(a)
                    FROM table1";

    let msg = format!("Creating logical plan for '{sql}'");
    let dataframe = ctx.sql(sql).await.expect(&msg);
    let physical_plan = dataframe.create_physical_plan().await?;
    let batches = collect(physical_plan, ctx.task_ctx()).await?;
    print_batches(&batches)?;
    Ok(())
}
```

Test below shows that datafusion has no problem with showing 0 columns.
```rust
#[tokio::test]
async fn test_empty_table() -> Result<()> {
    let config = SessionConfig::new()
        .with_target_partitions(1);
    let ctx = SessionContext::with_config(config);
    ctx.sql("CREATE TABLE empty_table ()").await?;

    let sql = "SELECT *
                    FROM empty_table";

    let msg = format!("Creating logical plan for '{sql}'");
    let dataframe = ctx.sql(sql).await.expect(&msg);
    let physical_plan = dataframe.create_physical_plan().await?;
    let batches = collect(physical_plan, ctx.task_ctx()).await?;
    print_batches(&batches)?;
    Ok(())
}
```

### Expected behavior

I expect first query to run successfully

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Jefffrey**
## Which issue does this PR close?

Closes #6510

## Rationale for this change

Add test for issue to ensure no regression

## What changes are included in this PR?

New sqllogictest case

### CODE REVIEW

**Review by Weijun-H — APPROVED** [review]
LGTM! Thanks @Jefffrey

File: datafusion/sqllogictest/test_files/select.slt [hunk 1]
```diff
@@ -848,6 +848,11 @@ statement error DataFusion error: Error during planning: EXCLUDE or EXCEPT conta
 SELECT * EXCLUDE(a, a)
 FROM table1
 
+# if EXCEPT all the columns, query should still succeed but return empty
```
Comment (comphead) [inline datafusion/sqllogictest/test_files/select.slt:851]: thanks @Jefffrey i would also add more tricky test cases, where column names are in different order, duplicated.
like
```
SELECT * EXCEPT(d, b, c, a, a, b, c, d)
FROM table1
```

**Comment by alamb** [comment]
Thank you @Jefffrey

**Review by alamb — APPROVED** [review]
I think this PR is better than master so we can merge it in. However I think @comphead 's ideas are also worth considering. We could do it as a follow on PR as well

File: datafusion/sqllogictest/test_files/select.slt (hunk 1 above)
Comment (Jefffrey) [reply datafusion/sqllogictest/test_files/select.slt:851]: Sure thing, added tests to cover this

**Review by alamb — APPROVED** [review]
Thanks again @Jefffrey

**Pull request merged**